minimum integer value causes errors

The following test fails in meeshkan because openapi-typed expects the minimum field to be float:
```python
def get_spec(schema):
    return convert_to_OpenAPIObject({"openapi": "3.0",
                                     "info": {"title": "Title", "version": "1.1.1"},
                                     "paths": {"/":
                                                   {"get": {"responses": {"200": {"description": "some",
                                                                                  "content": {"application/json": {"schema": schema}}}}}}}})

def test_faker_2():
    request = RequestBuilder.from_dict(dict(method="get",  protocol="http", path="/", host="api.com"))

    schema = {
        "$id": "https://example.com/person.schema.json",
        "$schema": "http://json-schema.org/draft-07/schema#",
        "title": "Person",
        "type": "object",
        "properties": {
            "firstName": {"type": "string", "description": "The person's first name."},
            "lastName": {"type": "string", "description": "The person's last name."},
            "age": {
                "description": "Age in years which must be equal to or greater than zero.",
                "type": "integer",
                "minimum": 0,
            },
        },
    }
    res = MeeshkanSchemaFaker(Faker(), request, get_spec(schema), storage_manager.default).execute()
    assert valid_schema(res.bodyAsJson, schema)
```

The exception is:
```
                except Exception as e:
>                   raise ValueError('%s\nKey %s has a value %s which cannot be cast to PathItem.' % (str(e), k, str(v)))
E                   ValueError: minimum must be of type `float` in Schema, instead got <class 'int'>
E                   Key / has a value {'get': {'responses': {'200': {'description': 'some', 'content': {'application/json': {'schema': {'$id': 'https://example.com/person.schema.json', '$schema': 'http://json-schema.org/draft-07/schema#', 'title': 'Person', 'type': 'object', 'properties': {'firstName': {'type': 'string', 'description': "The person's first name."}, 'lastName': {'type': 'string', 'description': "The person's last name."}, 'age': {'description': 'Age in years which must be equal to or greater than zero.', 'type': 'integer', 'minimum': 0}}}}}}}}} which cannot be cast to PathItem.

../../../anaconda3/envs/py36/lib/python3.6/site-packages/openapi_typed_2/converters.py:70: ValueError
```

--- a/gen.py
+++ b/gen.py
@@ -1,27 +1,24 @@
-import openapi_typed_2
-from collections.abc import Sequence, Mapping
-
 _TO_DICT_SUBS = {'_license': 'license', '_format': 'format', '_in': 'in', '_ref': '$ref', '_not': 'not', '_type': 'type', '_default': 'default'}
 
-def _simple_prop(n: str, q: str, K: str, tp: str) -> str:
+def _simple_prop(n: str, q: str, K: str, tp: str, ctp: str) -> str:
     O = '''    {}: {}
     if '{}' not in d:
         raise ValueError('{} must be defined for {}')
-    if not isinstance(d['{}'], {}):
-        raise ValueError('{} must be of type `{}` in {}, instead got %s' % str(type(d['{}'])))
-    {} = cast({}, d['{}'])
-'''.format(n, tp, q, q, K, q, tp, q, tp, K, q, n, tp, q)
+    if not iscoerceable(d['{}'], {}):
+        raise ValueError('{} must be one of the following types `{}` in {}, instead got %s' % str(type(d['{}'])))
+    {} = cast({}, {}(d['{}']))
+'''.format(n, tp, q, q, K, q, ctp, q, ctp, K, q, n, tp, tp, q)
     return O
 
-def _simple_prop_opt(n: str, q: str, K: str, tp: str) -> str:
+def _simple_prop_opt(n: str, q: str, K: str, tp: str, ctp: str) -> str:
     O = '''    {}: Optional[{}]
     if '{}' not in d:
         {} = None
-    elif not isinstance(d['{}'], {}):
-        raise ValueError('{} must be of type `{}` in {}, instead got %s' % str(type(d['{}'])))
+    elif not iscoerceable(d['{}'], {}):
+        raise ValueError('{} must be one of the following types: `{}` in {}, instead got %s' % str(type(d['{}'])))
     else:
         {} = cast({}, d['{}'])
-'''.format(n, tp, q, n, q, tp, q, tp, K, q, n, tp, q)
+'''.format(n, tp, q, n, q, ctp, q, ctp, K, q, n, tp, q)
     return O
 
 # Optional[Mapping[str, Union[Link, Reference]]]
@@ -398,28 +395,28 @@ def osupr(n: str, q: str, K: str):
     return _osu_r(n, q, K, 'Parameter')
 
 def sstr(n: str, q: str, K: str) -> str:
-    return _simple_prop(n, q, K, 'str')
+    return _simple_prop(n, q, K, 'str', '[str]')
 
 def sint(n: str, q: str, K: str) -> str:
-    return _simple_prop(n, q, K, 'int')
+    return _simple_prop(n, q, K, 'int', '[int]')
 
 def sbool(n: str, q: str, K: str) -> str:
-    return _simple_prop(n, q, K, 'bool')
+    return _simple_prop(n, q, K, 'bool', '[bool]')
 
 def sfloat(n: str, q: str, K: str) -> str:
-    return _simple_prop(n, q, K, 'float')
+    return _simple_prop(n, q, K, 'float', '[float, int]')
 
 def sstro(n: str, q: str, K: str) -> str:
-    return _simple_prop_opt(n, q, K, 'str')
+    return _simple_prop_opt(n, q, K, 'str', '[str]')
 
 def sinto(n: str, q: str, K: str) -> str:
-    return _simple_prop_opt(n, q, K, 'int')
+    return _simple_prop_opt(n, q, K, 'int', '[int]')
 
 def sboolo(n: str, q: str, K: str) -> str:
-    return _simple_prop_opt(n, q, K, 'bool')
+    return _simple_prop_opt(n, q, K, 'bool', '[bool]')
 
 def sfloato(n: str, q: str, K: str) -> str:
-    return _simple_prop_opt(n, q, K, 'float')
+    return _simple_prop_opt(n, q, K, 'float', '[float, int]')
 
 _CONVERTERS = {
     'float': sfloat,
@@ -492,6 +489,9 @@ from .openapi import *
 from dataclasses import is_dataclass, fields
 
 _TO_DICT_SUBS = {'_license': 'license', '_format': 'format', '_in': 'in', '_ref': '$ref', '_not': 'not', '_type': 'type', '_default': 'default'}
+
+def iscoerceable(i: any, lt: Sequence[type]) -> bool:
+    return True in [isinstance(i, x) for x in lt]
 
 def convert_to_str(d: Any) -> str:
     if not isinstance(d, str):

--- a/test/test_petstore_typing.py
+++ b/test/test_petstore_typing.py
@@ -37,7 +37,8 @@ meow = {
                         "schema": {
                             "type": "array",
                             "items": {
-                                "type": "string"
+                                "type": "float",
+                                "minimum": 0
                             }
                         }
                     },

